compiler/loader/unix_spec.cr fails on FreeBSD due to the different error message wording

Two expectations are currently failing:

spec/compiler/loader/unix_spec.cr:31:
```
Expected Crystal::Loader::LoadError with message matching /\/usr\/home\/user\/crystal\/foobar\.o.+(No such file or directory|image not found)/, got #<Crystal::Loader::LoadError: cannot load /usr/home/user/crystal/foobar.o (Cannot open "/usr/home/user/crystal/foobar.o")
```

spec/compiler/loader/unix_spec.cr:34:
```
Expected Crystal::Loader::LoadError with message matching /\/usr\/home\/user\/crystal\/foo\/bar\.o.+(No such file or directory|image not found)/, got #<Crystal::Loader::LoadError: cannot find -lfoo/bar.o (Cannot open "/usr/home/user/crystal/foo/bar.o")
```

--- a/spec/compiler/loader/unix_spec.cr
+++ b/spec/compiler/loader/unix_spec.cr
@@ -28,10 +28,10 @@ describe Crystal::Loader do
     end
 
     it "parses file paths" do
-      expect_raises(Crystal::Loader::LoadError, /#{Dir.current}\/foobar\.o.+(No such file or directory|image not found)/) do
+      expect_raises(Crystal::Loader::LoadError, /#{Dir.current}\/foobar\.o.+(No such file or directory|image not found)|Cannot open "#{Dir.current}\/foobar\.o"/) do
         Crystal::Loader.parse(["foobar.o"], search_paths: [] of String)
       end
-      expect_raises(Crystal::Loader::LoadError, /#{Dir.current}\/foo\/bar\.o.+(No such file or directory|image not found)/) do
+      expect_raises(Crystal::Loader::LoadError, /(#{Dir.current}\/foo\/bar\.o).+(No such file or directory|image not found)|Cannot open "#{Dir.current}\/foo\/bar\.o"/) do
         Crystal::Loader.parse(["-l", "foo/bar.o"], search_paths: [] of String)
       end
     end

